src/matrices.jl

```diff
@@ -0,0 +1,53 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+abstract type CoDaMatrix{T} end
+
+"""
+    JMatrix{T}
+
+Square matrix of ones.
+"""
+struct JMatrix{T} <: CoDaMatrix{T} end
+JMatrix{T}(d::Integer) where {T} = Ones{T}(d, d)
+(J::JMatrix{T})(d::Integer) where {T} = JMatrix{T}(d)
+const J = JMatrix{Bool}()
```

Comment:
I would move this const to after the section where you define the operations. You could even group all these instances at the bottom of the source file with a comment saying that these are instances exported for end-users so that they can write clean expressions like F'*T*F.